### ISSUE TITLE
Audio Crackle when browser source is set to "Control audio via OBS"

### ISSUE THREAD

**Issue by TheRealDadbeard** [obsproject/obs-studio#6612]
### Operating System Info

Windows 11

### OBS Studio Version

27.2.4

### OBS Studio Log URL

https://obsproject.com/logs/yGAxWFqdh0G7zqMf

### Expected Behavior

No audio crackle

### Current Behavior

Audio in browser source that have the option "Control audio via OBS" enabled have distorted/crackling sound when monitoring is enabled. Monitored sound will crackle but not the audio in the output.

### Steps to Reproduce

1. Add a browser source where you can play audio (SoundAlerts for example)
2. Enable "Control audio via OBS" and "Output and Monitoring"
3. Play a sound and hear it crackle (shorter ones seem to trigger it more)

### Anything else we should know?

I have this issue everytime, doesn't matter if only OBS is running or not. Maybe the new CEF version behaves differently on W11.

**Comment by MattyFresh68** [comment]
I'm on win11 and do not get audio distortion/crackle from browser sources. This has been reported, but I think it was specific to things by triggerfyre iirc

**Comment by skeletonbow** [comment]
Sounds most likely like it could be mismatched sample rate configuration in Windows, OBS or other parts of the audio stack.  This is the usual symptom of that.

**Comment by TheRealDadbeard** [comment]
> I'm on win11 and do not get audio distortion/crackle from browser sources. This has been reported, but I think it was specific to things by triggerfyre iirc

With "Control audio via OBS" and "Output and Monitoring"?

> Sounds most likely like it could be mismatched sample rate configuration in Windows, OBS or other parts of the audio stack. This is the usual symptom of that.

Nope. Everything is on 48khz

**Comment by prgmitchell** [comment]
I have reached out in the support channels to troubleshoot when available.

**Comment by pkviet** [comment]
Could you retry the repro steps and upload the log with a clean install without third party plugins ? You can use a portable install of obs to that purpose (download a zip, unzip wherever convenient and launch in portable mode).

**Comment by TheRealDadbeard** [comment]
> Could you retry the repro steps and upload the log with a clean install without third party plugins ? You can use a portable install of obs to that purpose (download a zip, unzip wherever convenient and launch in portable mode).

Yes I can do that later

**Comment by TheRealDadbeard** [comment]
Okay just did with a clean obs and the same happens.

**Comment by NarryG** [comment]
This is an issue that has been tracked in the past and while it's improved, it's never been fully resolved. 

There was a fairly long conversation in one of the beta channels in December where I was showing off these issues, but that channel has been archived/deleted from the Discord.  If it's in an archive state a Discord mod should be able to find extensive test-cases alongside audio clips that will cause consistent crackling.

This commit https://github.com/obsproject/obs-studio/pull/5648 helped substantially as it removed the crackling from the output, but monitoring will still crackle on occasion.

I'm still facing the same issues. There's large inconsistencies in how much it crackles and when it crackles.

**Comment by TheRealDadbeard** [comment]
> This is an issue that has been tracked in the past and while it's improved, it's never been fully resolved.
> 
> There was a fairly long conversation in one of the beta channels in December where I was showing off these issues, but that channel has been archived/deleted from the Discord. If it's in an archive state a Discord mod should be able to find extensive test-cases alongside audio clips that will cause consistent crackling. If it's been deleted, then it's just another victim of unarchived Discord channels destroying knowledge that could come in handy in the future and I can go digging for test cases again.
> 
> This commit #5648 helped substantially as it removed the crackling from the output, but monitoring will still crackle on occasion.
> 
> I'm still facing the same issues. There's large inconsistencies in how much it crackles and when it crackles.

Yeah I have had this issue for years, never to be completely fixed for me but it's very consistent for me with fast sounds. Thats why it is so noticeable for me when someone triggers a soundalert.

**Comment by NarryG** [comment]
> > This is an issue that has been tracked in the past and while it's improved, it's never been fully resolved.
> > There was a fairly long conversation in one of the beta channels in December where I was showing off these issues, but that channel has been archived/deleted from the Discord. If it's in an archive state a Discord mod should be able to find extensive test-cases alongside audio clips that will cause consistent crackling. If it's been deleted, then it's just another victim of unarchived Discord channels destroying knowledge that could come in handy in the future and I can go digging for test cases again.
> > This commit #5648 helped substantially as it removed the crackling from the output, but monitoring will still crackle on occasion.
> > I'm still facing the same issues. There's large inconsistencies in how much it crackles and when it crackles.
> 
> Yeah I have had this issue for years, never to be completely fixed for me but it's very consistent for me with fast sounds. Thats why it is so noticeable for me when someone triggers a soundalert.

The browser source also doesn't pass the the first (I believe) 3 packets to OBS. This was an issue discovered and once again is buried in the archived beta channel on Discord as it was never turned into an issue.  It appeared to be Chromium. Nobody could comment if it was related to that for sure, but we were unable to build a reliable test case beyond short audio clips for alerts were the worst. 

If any of the devs can comment if that channel still exists and is archived, I can turn the information we debugged then into proper issues if it can be made available.

**Comment by pkviet** [comment]
On Sun, Jun 12, 2022, 22:13 Dan (Narry) ***@***.***> wrote:

> This is an issue that has been tracked in the past and while it's
> improved, it's never been fully resolved.
> There was a fairly long conversation in one of the beta channels in
> December where I was showing off these issues, but that channel has been
> archived/deleted from the Discord. If it's in an archive state a Discord
> mod should be able to find extensive test-cases alongside audio clips that
> will cause consistent crackling. If it's been deleted, then it's just
> another victim of unarchived Discord channels destroying knowledge that
> could come in handy in the future and I can go digging for test cases again.
> This commit #5648 <https://github.com/obsproject/obs-studio/pull/5648>
> helped substantially as it removed the crackling from the output, but
> monitoring will still crackle on occasion.
> I'm still facing the same issues. There's large inconsistencies in how
> much it crackles and when it crackles.
>
> Yeah I have had this issue for years, never to be completely fixed for me
> but it's very consistent for me with fast sounds. Thats why it is so
> noticeable for me when someone triggers a soundalert.
>
> The browser source also doesn't pass the the first (I believe) 3 packets
> to OBS. This was an issue discovered and once again is buried in the
> archived beta channel on Discord as it was never turned into an issue. It
> appeared to be Chromium. Nobody could comment if it was related to that for
> sure, but we were unable to build a reliable test case beyond short audio
> clips for alerts were the worst.
>

I confirmed the issue and we're well aware of it.
Some recent chromium changes with audio will perhaps help.

> If any of the devs can comment if that channel still exists and is
> archived, I can turn the information we debugged then into proper issues if
> it can be made available.
>
The beta channel still exists.
Ping Fenrir , Rytoex or MattWizardCM on discord about it to see what they
think.

**Comment by TheRealDadbeard** [comment]
Would be great to see this finally fixed

**Comment by TheRealDadbeard** [comment]
Some new information from me:

I moved some of my audio alerts over to simple media sources that get triggered by events. I noticed the same crackle appers on them too. Any source that seem to play a sound immediately it becomes active will crackle. There is no difference in what plugins are installed or the newest beta.

It crackles every single time but as always only for monitoring. Output is clean like 99%, very seldom a crackle there.

**Comment by Nikoheartttv** [comment]
Happens to me as well. Trying to pass through Text-to-speech through the Control Audio via OBS and it always initially crackles. Goes for any sound to be honest

**Comment by Nikoheartttv** [comment]
Just wanted to bump this, this is still happening in OBS 28.0.2

**Comment by maelstromALPHA** [comment]
I have had this issue for years now and I'm still getting audio popping/crackling whenever a sound initially plays from a browser source as of the most recent version. Hope to see a fix soon!

**Comment by prenetic** [comment]
Same thing here with 28.0.3, have been pulling my hair out trying to determine the cause but this is exactly the scenario where it's occurring. Browser sources that are set to "Control audio via OBS" since I don't have the option to adjust levels at the source itself and need to reduce the gain.

**Comment by gnkidwell** [comment]
Having this exact same issue with OBS 28.0.3… Any browser source with Control Audio via OBS is creating choppy audio… with that unchecked no issues with any of the browser sources…
